Content.Shared/SpittableContainer/SharedSpittableContainerSystem.cs

```diff
@@ -0,0 +1,136 @@
+using Content.Shared.Actions;
+using Content.Shared.DoAfter;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Item;
+using Content.Shared.Popups;
+using Content.Shared.SpittableContainer.Components;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Containers;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.SpittableContainer;
+
+public abstract class SharedSpittableContainerSystem : EntitySystem
+{
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    [Dependency] protected readonly SharedContainerSystem _containerSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SpittableContainerComponent, ComponentInit>(OnInit);
+        SubscribeLocalEvent<SpittableContainerComponent, ComponentShutdown>(OnShutdown);
+        SubscribeLocalEvent<SpittableContainerComponent, SwallowToContainerActionEvent>(OnSwallowToContainerAction);
+        SubscribeLocalEvent<SpittableContainerComponent, SwallowDoAfterEvent>(OnSwallowDoAfter);
+        SubscribeLocalEvent<SpittableContainerComponent, SpitFromContainerActionEvent>(OnSpitFromContainerAction);
+        SubscribeLocalEvent<SpittableContainerComponent, SpitFromContainerDoAfterEvent>(OnSpitDoAfter);
+    }
+
+    private void OnInit(Entity<SpittableContainerComponent> ent, ref ComponentInit args)
+    {
+        ent.Comp.Container = _containerSystem.EnsureContainer<Container>(ent.Owner, ent.Comp.Storage);
+
+        AddActionIfNeeded(ent.Owner, ref ent.Comp.SwallowActionEntity, ent.Comp.SwallowActionPrototype);
+        AddActionIfNeeded(ent.Owner, ref ent.Comp.SwallowActionEntity, ent.Comp.SpitContainerActionPrototype);
```

Comment:
First of all, you're setting the SwallowActionEntity twice.

Instead of calling this function, can't you just add the actions here directly?

```suggestion
        _actionsSystem.AddAction(ent, ref ent.Comp.SwallowActionEntity, ent.Comp.SwallowActionPrototype);
        _actionsSystem.AddAction(ent, ref ent.Comp.SpitContainerActionEntity, ent.Comp.SpitContainerActionPrototype);
```